MatthewKangA6

```diff
@@ -0,0 +1,159 @@
+#include <vector>
+using std::vector;
+
+
+#include <iostream>
+using std::ostream;
+using std::cout;
+using std::endl;
+
+
+
+void print_array(int i, const int a[]) {
+	cout << "( ";
+
+	for (int n = 0; n < i; n++) {
+		cout << a[n] << " ";
+	}
+	cout << ")";
+
+}
+
+void selection_sort_in_place(int i, int a[]) {
+	for (int index = 0; index < i; index++) {
+		int finalIndex = index;
+		for (int compare = index; compare < i; compare++) {
+			if (a[compare] < a[index])
+				if (a[compare] < a[finalIndex])
```

Comment:
unnecessary to do both these tests -- `if (a[compare] < a[finalIndex])` should suffice